src/mantine-hooks/src/use-viewport-size/use-viewport-size.ts

```diff
@@ -0,0 +1,36 @@
+import { useCallback, useState, useEffect } from 'react';
+import { useWindowEvent } from '../use-window-event/use-window-event';
+
+const eventListerOptions = {
+  passive: true,
+};
+
+export function useViewportSize(): Readonly<{
+  width: number;
+  height: number;
+}> {
+  const [windowSize, setWindowSize] = useState({
+    width: 0,
```

Comment:
Can we initialize these values in browser in useState instead of useeffect?

```tsx
const [windowSize, setWindowSize] = useState({
  width: typeof window !== 'undefined' ? window.innerWidth : 0,
  height: typeof window !== 'undefined' ? window. innerHeight : 0,
});
```

This way we can save up setting state after component was mounted